sisl/geom/nanoribbon.py

```diff
@@ -154,3 +159,533 @@ def zgnr(width, bond=1.42, atoms=None):
     agnr : armchair graphene nanoribbon
     """
     return graphene_nanoribbon(width, bond, atoms, kind='zigzag')
+
+
+_heteroribbon_section = namedtuple("Section", ["W", "L", "shift", "align", "atoms"], defaults=(1, 0, None, None))
+
+
+@set_module("sisl.geom")
+def heteroribbon(sections, bond, atoms, kind="armchair",
+                 invert_first=False, align="bottom",
+                 correct_shifts=False, on_dangling="raise", pbc=True):
```

Comment:
What about `low_coordination="warn"` as default (replacing `on_dangling`), referring to creation of structures with neighbours < 2?

Maybe `correct_shifts` could be a function of the above parameter:
- `low_coordination="raise"` => `correct_shifts=True`
- `low_coordination="warn/ignore"`=> `correct_shifts=False` (the user knows what they are doing)?

Reply:
I think you should be able to provide the exact shift yourself and still get the function to error if you get a single bonded atom. I'm thinking mainly if you do high throughput and you know you don't want single bonded edges, but because of some bug in your code you are producing the wrong structures.

Reply:
One could also have a parameter `min_coordination=2`, i.e., that the user specifies the minimum coordination that can be tolerated?